lib/ansible/executor/module_common.py

```diff
@@ -159,7 +159,7 @@ def invoke_module(module, modlib_path, json_params):
     else:
         os.environ['PYTHONPATH'] = modlib_path
 
-    p = subprocess.Popen([%(interpreter)s, module], env=os.environ, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE)
+    p = subprocess.Popen([%(interpreter)s, '-m', module], env=os.environ, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE)
```

Comment:
Talked about this on slack and decided not too.

(1) Then it only works for ansiballz, nothing else (for instance, can't fix php).
(2) it introduces parsing and scraping into the wrapper which is a bad environment for debugging anything.
(3) the problem lies with the ssh connection plugin deciding that error code 255 is special so we have to teach the ssh connection plugin when error 255 is not special.

Reply:
its not the ssh connection plugin thinking that, it is ssh itself, the plugin just follows what the cli tool does.

 I don't see why you need parsing since we already capture the exit code.